drivers/temperature/adt7420/adt7420.c

```diff
@@ -92,38 +90,68 @@ uint16_t adt7420_get_register_value(struct adt7420_dev *dev,
  *
  * @param dev              - The device structure.
  * @param register_address - Address of the register.
- * @param num_data_bytes   - Number of bytes.
- * @param data             - Data to be written in input register.
  * @param mask		   - Bit Mask of the bit to be written
  * @param value		   - Value of the bit
  *
  * @return 0 in case of success, -1 otherwise.
 *******************************************************************************/
-int32_t set_register_value(struct adt7420_dev *dev,
-			   uint8_t register_address,
-			   uint8_t num_data_bytes,
-			   uint8_t *data, uint8_t mask, uint8_t value)
+int adt7420_set_register_mask(struct adt7420_dev *dev,
+			      uint8_t register_address, uint8_t mask, uint8_t value)
 {
 
 	uint8_t data_buffer[3] = {0, 0, 0};
+	uint16_t regval;
+
+	regval = adt7420_get_register_value(dev, register_address);
+
+	regval &= ~mask;
+	regval |= value;
+	if (!adt7420_is_spi(dev) || register_address == ADT7320_REG_STATUS
```

Comment:
This approach looks a bit odd. Since we are updating the driver, i'd suggest a more common approach (used also in the linux drivers) when registers have variable data size.

We should somehow store the data size of the registers in the macros:
https://github.com/analogdevicesinc/no-OS/blob/9f4b208d4b411b64707a8beed4cb0bac5dbbeace/drivers/dac/ad7293/ad7293.h#L56-L68

And process them accodingly in the spi read/write functions.